tests/test_mql.py

```diff
@@ -1001,3 +1105,183 @@ def test_parse_mql_arbitrary_functions(
+            )
+        ),
+        id="test curried arbitrary function with filter",
+    ),
+    pytest.param(
+        "topK(10)(transaction.duration{bar:baz} by transaction)",
+        MetricsQuery(
+            query=Timeseries(
+                metric=Metric(public_name="transaction.duration"),
+                aggregate="topK",
+                aggregate_params=[10],
+                filters=[Condition(Column("bar"), Op.EQ, "baz")],
+                groupby=[Column("transaction")],
+            )
+        ),
+        id="test curried arbitrary function with filter and groupby",
+    ),
+    pytest.param(
+        "topK(10)(transaction.duration)",
+        MetricsQuery(
+            query=Timeseries(
+                metric=Metric(public_name="transaction.duration"),
+                aggregate="topK",
+                aggregate_params=[10],
+            )
+        ),
+        id="test curried arbitrary function with args",
+    ),
+    pytest.param(
+        'topK(10)("test.duration")',
+        MetricsQuery(
+            query=Formula(
+                function_name="topK",
+                aggregate_params=[10],
+                parameters=["test.duration"],
+            )
+        ),
+        id="test curried arbitrary function with string param",
+    ),
+    pytest.param(
+        "topK(10)(sum(transaction.duration))",
+        MetricsQuery(
+            query=Formula(
+                function_name="topK",
+                aggregate_params=[10],
+                parameters=[
+                    Timeseries(
+                        metric=Metric(public_name="transaction.duration"),
+                        aggregate="sum",
+                    ),
+                ],
+            )
+        ),
+        id="test curried arbitrary function with inner aggregate",
+    ),
+    pytest.param(
+        "topK(10)(sum(transaction.duration) / count(transaction.duration))",
+        MetricsQuery(
+            query=Formula(
+                function_name="topK",
+                aggregate_params=[10],
+                parameters=[
+                    Formula(
+                        function_name="divide",
+                        parameters=[
+                            Timeseries(
+                                metric=Metric(public_name="transaction.duration"),
+                                aggregate="sum",
+                            ),
+                            Timeseries(
+                                metric=Metric(public_name="transaction.duration"),
+                                aggregate="count",
+                            ),
+                        ],
+                    ),
+                ],
+            )
+        ),
+        id="test curried arbitrary function with inner aggregate and terms",
+    ),
+    pytest.param(
+        "topK(10)(sum(transaction.duration{bar:baz}) / count(transaction.duration{foo:foz})) by transaction",
+        MetricsQuery(
```

Comment:
Could you add some of these more complex examples to the serializing tests as well? `test_metrics_mql_query.py`? Just to make sure we are properly serializing these things.

Reply:
yes, I can add this 👍